ui/src/pages/backoffice/Acquisition/Order/OrderEditor/components/OrderForm/OrderForm.js

```diff
@@ -34,6 +25,11 @@ const orderSubmitSerializer = (values, newRecord) => {
         ? line.document.id
         : line.document.pid;
     }
+    if (line.patron) {
+      line.patron_pid = _has(line.patron, 'id')
+        ? line.patron.id
+        : line.patron.pid;
```

Comment:
If sometimes we have `id` and sometimes `pid`, we might have to fix this... did you notice that we have both or it is just to be safe?

Reply:
So when we use the `ESSelector` it uses this [serializer](https://github.com/inveniosoftware/invenio-app-ils/blob/master/ui/src/components/ESSelector/serializer.js) which uses `id` because not everything searchable is a record. The existing rendered data are not getting serialized (so they have pids) only the search results. We should address it in an application scale level. Thanks to @equadon for helping with the struggle on this one particularly.